src/popup/components/SwapRates.vue

```diff
@@ -37,39 +37,44 @@
   </div>
 </template>
 
-<script>
-import { mapState } from 'vuex';
+<script lang="ts">
+import { computed, defineComponent } from '@vue/composition-api';
 import { camelCase } from 'lodash-es';
 import { transactionTokenInfoResolvers } from '../utils/transactionTokenInfoResolvers';
 import { FUNCTION_TYPE_DEX } from '../utils/constants';
+import { useFungibleTokens } from '../../composables';
 
 import Tokens from './Tokens.vue';
 import TokenAmount from './TokenAmount.vue';
 
-export default {
+export default defineComponent({
   components: {
     Tokens,
     TokenAmount,
   },
   props: {
     transaction: { type: Object, required: true },
   },
-  computed: {
-    ...mapState('fungibleTokens', ['availableTokens']),
-    isSwapTx() {
-      return [
-        ...FUNCTION_TYPE_DEX.swap,
-        ...FUNCTION_TYPE_DEX.pool,
-      ].includes(this.transaction.tx.function);
-    },
-    rates() {
-      if (!this.isSwapTx) return [];
+  setup(props, { root }) {
+    const { availableTokens } = useFungibleTokens({ store: root.$store });
+
+    const isSwapTx = computed(() => ([
+      ...FUNCTION_TYPE_DEX.swap,
+      ...FUNCTION_TYPE_DEX.pool,
+    ].includes(props.transaction.tx.function)));
 
-      const resolver = transactionTokenInfoResolvers[camelCase(this.transaction.tx.function)];
+    const rates = computed(() => {
+      if (!isSwapTx.value) {
+        return [];
+      }
+
+      const resolver = (transactionTokenInfoResolvers as any)[
```

Comment:
Please don't add `as any` to `transactionTokenInfoResolvers`. I've made big effort to type this. If you have issues with typing it I suggest to use Paweł's solution done in `transactionTokens.ts` file:
```
const functionName = camelCase(transaction.tx.function) as TxFunctionParsed;
```